File: combat/src/classicthunder/combat/cards/StrikeCard.java
package classicthunder.combat.cards;

import classicthunder.combat.combat.character.CharacterStats;
import classicthunder.combat.combat.character.CharacterType;
import com.badlogic.gdx.graphics.Texture;

public class StrikeCard extends Card {

    public StrikeCard(Texture texture) {
        super(texture, 1);
    }

    @Override
    public int GetEnergyCost() {
        return 1;
    }

    @Override
    public boolean CanApplyEffects(CharacterStats stats) {

        return (stats.getCharacterType() == CharacterType.ENEMY);
    }

    @Override
    public void ApplyEffects(CharacterStats stats) {
        stats.AdjustHealth(-6);
    }
}

File: deck/src/classicthunder/card/StrikeCard.kt
package classicthunder.card

import classicthunder.character.CharacterStats
import classicthunder.character.CharacterType
import com.badlogic.gdx.graphics.Texture

class StrikeCard(texture: Texture) : Card(texture)
{
    override fun getEnergyCost(): Int {
        return 1
    }

    override fun canApplyEffects(stats: CharacterStats): Boolean {
        return stats.characterType == CharacterType.ENEMY
    }

    override fun applyEffects(stats: CharacterStats) {
        stats.AdjustHealth(-6)
    }
}